src/ReaderScreen/utils/index.js

```diff
@@ -1,5 +1,186 @@
+  fontSize,
+  isNightMode,
+  isLarivaar,
+  punjabiTranslation = ""
+) => {
+  const fontClass =
+    type === constant.GURMUKHI.toLowerCase() || punjabiTranslation !== ""
+      ? constant.GURMUKHI.toLowerCase()
+      : type;
+  return `
+    <div class="content-item ${fontClass} ${textAlign}" style="font-size: ${fontSizeForReader(
+    fontSize,
+    header,
+    type === constant.TRANSLITERATION.toLowerCase() || type === constant.TRANSLATION.toLowerCase(),
+    isLarivaar
+  )}px; color: ${fontColorForReader(header, isNightMode, type.toUpperCase())};">
+      ${content}
+    </div>
+  `;
+};
+
+export const loadHTML = (
+  shabadID,
+  shabad,
+  isTransliteration,
+  fontSize,
+  fontFace,
+  isEnglishTranslation,
+  isPunjabiTranslation,
+  isSpanishTranslation,
+  isNightMode,
+  isLarivaar,
+  savePosition
+) => {
+  try {
+    const backColor = isNightMode ? colors.NIGHT_BLACK : colors.WHITE_COLOR;
+    const fileUri = Platform.select({
+      ios: `${fontFace}.ttf`,
+      android: `file:///android_asset/fonts/${fontFace}.ttf`,
+    });
+
+    const content = shabad
+      .map((item) => {
+        const textAlignMap = {
+          0: "left",
+          1: "center",
+          2: "center",
+        };
+
+        let textAlign = textAlignMap[item.header];
+        if (textAlign === undefined) {
+          textAlign = "right";
+        }
+        let contentHtml = `<div id="${item.id}" class='text-item'>`;
+        contentHtml += createDiv(
+          item.gurmukhi,
+          item.header,
+          constant.GURMUKHI.toLowerCase(),
+          textAlign,
+          fontSize,
+          isNightMode,
+          isLarivaar
+        );
+
+        if (isTransliteration) {
+          contentHtml += createDiv(
+            item.translit,
+            item.header,
+            constant.TRANSLITERATION.toLowerCase(),
+            textAlign,
+            fontSize,
+            isNightMode,
+            isLarivaar
+          );
+        }
+
+        if (isEnglishTranslation) {
+          contentHtml += createDiv(
+            item.englishTranslations,
+            item.header,
+            constant.TRANSLATION.toLowerCase(),
+            textAlign,
+            fontSize,
+            isNightMode,
+            isLarivaar
+          );
+        }
+
+        if (isPunjabiTranslation) {
+          contentHtml += createDiv(
+            item.punjabiTranslations,
+            item.header,
+            constant.TRANSLATION.toLowerCase(),
+            textAlign,
+            fontSize,
+            isNightMode,
+            isLarivaar,
+            constant.GURMUKHI.toLowerCase()
+          );
+        }
+
+        if (isSpanishTranslation) {
+          contentHtml += createDiv(
+            item.spanishTranslations,
+            item.header,
+            constant.TRANSLATION.toLowerCase(),
+            textAlign,
+            fontSize,
+            isNightMode,
+            isLarivaar
+          );
+        }
+
+        contentHtml += `</div>`;
```

Comment:
Can we not use JSX to build this markup, instead of doing it manually from strings?

Reply:
nopes veer ji as we tried multiple ways as we discussed earlier and we finalized this way.